### ISSUE TITLE
Notes on an install over the past few days

### ISSUE THREAD

**Issue by Dragon-32** [rfuzzo/MorrowindPlusPlus#3]
Been working my way through this after a recommendation from Danae. Excellent work all round, thanks for producing it.

I see there have been some updates to the guide so some of these maybe deprecated. After writing this out I intend to go through the guide again but not sure when that will be.

- [ ] **Services Restored** - Told to edit the plugin with TESAME but that's the first time that tool's been introduced. Link 404s
- [ ] **Cutscenes Revamped - Cavern of the Incarnate** - I had to create a Video folder for the .bik
- [ ]  **Ownership Indicator** - Had to set Data Files manually
- [ ]  **Alchemical Knowledge** - This mod requires the Game mechanics/Healthy appetite feature to be enabled in the Morrowind Code Patch. But the Google Docs sheet says to disable that option
- [ ] **Chargen Revamped** - Had to set Data Files manually
- [ ] **magicka Mastery MWSE** - Had to create folder structure and re-zip
- [ ] **Locks and Traps Detection** - This mod requires the Hidden locks and Hidden traps features to be enabled in the Morrowind Code Patch. Disabled on Google docs
- [ ] **Service Requirements Revised** - Only need to install file under Optional Files, that's 1.1
- [ ] **RoHT** - Says it's a FOMOD installer, looks like a BAIN to me.
- [ ] **Hlaalu Rank Adjust** - by Ithilien87 not Caeris
- [ ] **Fighters Guild Improved** - Says it's a FOMOD installer, looks like a BAIN to me.
- [ ] **Mage Robes** - Says it's a FOMOD installer, looks like a BAIN to me.
- [ ] **Love in the Time of Daedra** - links to RandomPal's assorted mods page with a TR-compatible version of the mod. Need [the original](https://www.nexusmods.com/morrowind/mods/43635) too. Need to install just the non-abot .esp from the original
- [ ] **Master of Puppets** - 	Download both the main mod and the Better Typography add-on. Use only .esp from the add-on
- [ ] **Divine Dagoths** - Config for plugins says to leave all enabled. I think should change Available ESPs so just: _Bob's Diverse Dagoths - Hostile Gilvoth.esp_ (matches PfP) and _Unique Dagoth Brandy & Amulet.ESP_
- [ ] **MCAR** - I think we're supposed to download "MCAR plus swim animation"? That needs MCP's "First person swim animations" enabled. And Google sheet says to enable that one. But the BAIN options don't match the webpage.
- [ ] **Indexes of Morrowind** - Had to set Data Files manually
- [ ] **Sheep-no-more** - Shown as Redundant in MO when AURA installed. AURA replaces the same audio file this does.
- [ ] **AURA** - I noted this had updated to 4.0.3 on June 15th, not sure on the relevance of that.
- [ ] **Creature VFX restoration** - Is in the Audio section. Should it be? Seems to be overriding some files so maybe it's because of required mod install order..? Maybe add some words as to why here?
- [ ] **Spriggans Glow** - Looks like a FOMOD installer
- [ ] **Familiar Faces by Caleb** - _(optional) Redguard Female Teeth Texture Fix_ Delete this folder? Maybe just my love of order being offended by this ending up in Data Files
- [ ] **Watch the Skies** - Updated	13th June 2022. Options for the rain mesh replacer seem to have been renamed
- [ ] **Complete Armor Joints** - Kahkahra's original flagged as redundant after installing the other one.
- [ ] **Arukinns Better Books and Scrolls** - Formatting issue?
	`Note that this mod contains lore-unfriendly textures for the books' pages. You can easily hide these textures, named Tx\_book\_pages\_.`
Rendering error? Trying to be an escape character?
Tx_book_pages_01.dds thru Tx_book_pages_11.dds
- [ ] **I Lava Good Mesh Replacer** - Saw this new mod, maybe relevant: [I Lava Good Intelligent Textures Compatibility Patch - Fix](https://www.nexusmods.com/morrowind/mods/51521)
- [ ] **Prerelease Prisonmoon** - Above this is the Nexus Mods URL of Baar Dau, looks like a rogue edit and could be removed
- [ ] **Big Imperial Halls** - Is "Imp_Big_4way.ESP" needed? I think this is a modder's resource .esp
- [ ] **Melchior's Excellent Grazelands Acacia** - Doesn't specify options to install. I think both are warranted.
		00 Core
		01 OAAB Saplings Patch
- [ ] **Better Dwemer Interiors** - "Better Dwemer Interiors Master (Update files)" - download is actually in Main Files section
- [ ] **Windmill in Caldera by ayse** - Wow, now that's a mod and a name from the past.
- [ ] **More believable bandit camps - an addon for Ashfall**. Has BAIN options
		00 Core
		01 TOTSP Addon
- [ ] **OAAB Pomegranates** - Has BAIN options
		00 Core
		01 WWGTMA Patch [We will go to mount Assarnibibi]
- [ ] **Anna-Sera** - This mod requires the Beta/actor collision box fix feature to be enabled in the Morrowind Code Patch. Not recommended on the Google Docs sheet
- [ ] **BCOM - Patches** - Editing the plugins so only "abotGondoliers_RVP.esp" gets installed feels very wrong. Is this _really_ correct? Maybe some explanation?
- [ ] **Finish** - "Run MWSE-Update from your game folder." - But _not_ from within Mod Organizer 2? Earlier said to always run via Mod Organizer 2 and screenshot of Tools set-up has a link for MWSE-Update.
- [ ] **Finish - Wrye Mash** - _HircineQuest_VCW.esp_ has a master of "VSW_Tomb_of_the_Snow_Prince.esm". Shouldn't we change to "Solstheim Tomb of The Snow Prince.esm" rather than just remove? That's what I've done as I assume there's some STotSP objects referenced in the plugin
- [ ] **Finish - tes3cmd** - Why not clean the plugins? Would need to exclude grass mods (just for speed) and Patch for Purists (fix a Bloodmoon error with Steel cuirass). 75 I have installed have some kind of junk in. Add running Fixit in Mash to the Instructions?
- [ ] **Finish - TES3merge** - shouldn't run TES3Merge *before* mlox so Merged Objects.esp is moved in load order?
- [ ] **MGE XE - distant land** - Errors generating distant statics. Missing textures:
		WL_wtx_vurt_pine_bark_01.dds
		wl_wtx_vurt_pine_grn_01.dds
		wl_wtx_vurt_pine_grn_02.dds
		wl_wtx_vurt_pine_grn_03.dds
		wl_wtx_vurt_pine_half_01.dds
		wl_wtx_vurt_pine_snow_01.dds
		They're all here:
		VSW_Solstheim_Graphical_Replacer_V2.1.0-46810-2-1-0-1584270620.7z\010 Solstheim - HD Worldspace Graphical Replacer\Textures\wl\F\
- [ ] **MGE XE - shaders** - Skoomaesthesia's shader isn't added to the shader chain, not sure if that's intentional
- [ ] **Running the game for the first time** - I think a section that pulls out the various MCM recommendations from the previous entries would be good. The ones I noted and made:
		**Improved Vanilla Levelling** - DISABLE Retroactive Health Calculation
		**Map and Compass** - Need to select map in MCM
		**PvP** - Increase attack frequency of enemies -> No

- [ ] **In game, not MCM, option** - _Character Sound Overhaul_ - The author recommends you set the Footsteps volume to minimum in your in-game audio settings.

Finally, I saw this mod. It seemed like a good idea..? [Tamriel Rebuilt - Original Soundtrack](https://www.nexusmods.com/morrowind/mods/47254).

**Comment by rfuzzo** [comment]
wow :D that's so many good notes. Thank you so much!!
I'll work them in

**Comment by rfuzzo** [comment]
> Finish - TES3merge - shouldn't run TES3Merge before mlox so Merged Objects.esp is moved in load order?

no, that would be bad. mlox needs to run before so that tes3merge merges the plugins correctly. I added tes3merge exclusions for the esps after Merged Objects - with that Merged Objects is fine at the end.

**Comment by rfuzzo** [comment]
> Finish - tes3cmd - Why not clean the plugins? Would need to exclude grass mods (just for speed) and Patch for Purists (fix a Bloodmoon error with Steel cuirass). 75 I have installed have some kind of junk in. Add running Fixit in Mash to the Instructions?

yeah I did that, but cleaning BCOM broke some MVRFs... so I'm a bit hesitant now with the cleaning.

**Comment by rfuzzo** [comment]
> Finish - "Run MWSE-Update from your game folder." - But not from within Mod Organizer 2? Earlier said to always run via Mod Organizer 2 and screenshot of Tools set-up has a link for MWSE-Update.

NullCascade recently updated MWSE-Update to not work from within MO2 (because of some permission problems people had). I need to update Tools setup. thanks

**Comment by rfuzzo** [comment]
> BCOM - Patches - Editing the plugins so only "abotGondoliers_RVP.esp" gets installed feels very wrong. Is this really correct? Maybe some explanation?
It's the other way round :D I'll update the list to make it more clear

**Comment by Dragon-32** [comment]
Glad that was useful

I started a new game and the Firemoth ship was visible, according to [PikachunoTM](https://forums.nexusmods.com/index.php?/topic/11393923-bethesda-official-plugins-naturalized/#entry109242208) should enable entertainers, bcsounds, and LeFemm from UMOPP rather than the merged plugin when running with the naturalized ones; for now at least.

Haven't checked but people say that fixes things. I'll edit this when I get a chance.

**Comment by rfuzzo** [comment]
alright! that's good to know!

**Comment by Dragon-32** [comment]
Settings things up as PikachunoTM suggested did indeed remove the Firemoth ship from chargen.

It could be worth telling people to clean the Boots of Blazing Speed plugin (BOBS.esp) with tes3cmd as that mod's got the evil GMSTs.

I've been getting an error message on game load due to Lyithdonea (MD_Azurian Isles.esm) that I'd downloaded from MD's Github
[i]Expression Error Unable to find object "az_heal_bandage_01" in script az_scr_heal_removal.[/i]
I think I'm just going to remove the mod as I'm repeatedly hitting the 32bit limit and getting CTDs.

Is it OK me just adding things to this one Issue, or would you prefer it if I created new ones?

**Comment by rfuzzo** [comment]
hej, thank you so much for your comments. :) I don't mind either way.

yeah I got the `az_heal_bandage_01` error as well, probably best to remove Lyithdonea then I guess. I'll ask MD about it
and for crashing, hmm I didn't get any so far, but if removing Lyithdonea helps in this department too...

cleaning, yeah. I added a section telling how to clean all esps. but it would probaly make sense to specify which *not* to clean (base esps, BCOM...)

**Comment by rfuzzo** [comment]
okay, so:

- [x] removed Lyithdonea
- [ ] fix the install instructions for UMOPP
- [x] add cleaning instructions

**Comment by Dragon-32** [comment]
I'd say to add _Fixes for Purists_ to the "don't clean" list too. That will intentionally have the same values as Tribunal for the Steel Cuirass. Tribunal.esm fixed the Morrowind.esm record and then Bloodmoon.esm has the original, wrong values from Morrowind.esm

**Comment by Dragon-32** [comment]
Hey. Me again. 

**BCoM - Patches**
I think the _abotGondoliers_RVP.esp_ is needed

I tried taking a gondolier (without this plugin) from Vivec's temple to Ebonheart and the route passed through the docs that BCoM adds. I tried it again with this mod enabled and the gondolier took a detour around them. I think that's what's meant by "activate both esp"

**Comment by rfuzzo** [comment]
Hi Dragon :D thanks!

> Bcom Patches
Interesting... I'll investigate

> Repairing Saves
Yeah I noticed that too... 🤔 I didn't really find anything useful either but I should look again. Hmm. I have to confess I never really used repair all, but it uses tes3cmd right? I could debug that...

**Comment by Dragon-32** [comment]
> Hi Dragon :D thanks!
> 
> > Bcom Patches
> 
> Interesting... I'll investigate
> 
> > Repairing Saves
> 
> Yeah I noticed that too... 🤔 I didn't really find anything useful either but I should look again. Hmm. I have to confess I never really used repair all, but it uses tes3cmd right? I could debug that...

It's solely a Mash thing. IIRC, it's where a record in a plugin doesn't have a valid mod index for a particular reference. That mod index should be the .esp or one of its .esm masters. It's been some years though. There's probably more in some of Wrye's documentation, he was pretty good about that.

I'd say maybe just have some words in the Repairing Saves section that it might highlight some errors with plugins not noticed until now. People should use Repair All on the plugins Mash highlights until they're all done and the person can Repair their save.

**Comment by rfuzzo** [comment]
alright, that's good advice (dint' know that myself)

**Comment by Dragon-32** [comment]
Not sure if I should have posted this on Discord or here, went with continuing here. 

**UI Expansion Better Training Icons** 
This is in the _UI > Fonts and Icons_ section of the guide. _UI Expansion_ itself is in the _UI > Gameplay_ section of the guide and installed afterwards.
With that install order, from what I can see in Mod Organizer 2, none of the assets from _UI Expansion Better Training Icons_ are actually used in game.

**Memory usage**
Think I mentioned this earlier but holy heck a modded Morrowind struggles with the 32bit limit.

Have you considered a "dealing with 32bit memory limits" section? When I first started out with this guide I was getting CTDs  multiple times in the hour. Now I'm down to about 1 an hour, depending on what I'm doing.

Things I've done:
_MGE XE distant land_
World texture size kept at default (both colour and normal map)
Distant static textures set to 1/8

_morrowind.ini settings_
Interior and exterior cell buffers set to 0
DontThreadLoad set to 1 (this maybe a placebo but I think it helps)

_external programs_
configured an exception in my AV for Morrowind directory
close RivaTuner Statistics Server before starting Morrowind
According to Sysinternal's VMMap these were accounting for ~40MB each of morrowind.exe's RAM usage

I've also got Process Lasso purging working sets and standby lists every few minutes (CleanMem can do this too) but I can't say I've seen any effect of that using Memory Monitor in game

Finally I've run textures through [Optimizer Textures](https://www.nexusmods.com/skyrim/mods/12801), which is a bit of a fiddle to configure with its haphazard approach to its own configuration options.

I think having _something_ here would be useful. I was expecting things to be less crash-tastic than they were when I finished the guide. It's still Morrowind but things are somewhat better now with those changes.

**Comment by rfuzzo** [comment]
Cool, thank you! I added a Memory Management section, that's a good idea. 👍 

also fixed UI expansion derp.

**Comment by Dragon-32** [comment]
Hey. Back here again as the Issue is still open.

Saw this in Hrnchamd's [sticky post](https://forum.nexusmods.com/index.php?showtopic=472884/#entry123079134) for MGE XE:

> NOTE: for users of DXVK: DXVK 2.1 has a bug with lighting that causes some scenes to be black. e.g. Head choice window and inventory portrait.
> Please use DXVK 2.0 until it gets fixed.

DXVK in Setup section links to relases/latest so that's 2.1. The async one doesn't seem to have a version based on 2.1 yet.

**Comment by rfuzzo** [comment]
ah right thanks, I downloaded 2.0, but didn't mention it on the guide
